File: forumClient/src/Dto/app/BoardInfo.ts
import type { TopicBasicInfo } from "./TopicInfo"
export type BoardEntry = {
    boardId: string,
    categoryId: string,
    boardName: string,
    boardDescription: string,
    createdByUserId: string
}

export type BoardBasicInfo = {
    board: BoardEntry, 
}

export type BoardFullInfo = {
    board: BoardEntry,
    topics: TopicBasicInfo[],
    totalTopics: number
}

export type CreateBoardRequest = {
    boardName: string,
    boardDescription: string,
    categoryId: string,
}

export type BoardSummary = {
    totalBoards: number
}

File: server/RestApiServer/Dto/App/BoardInfo.cs
using RestApiServer.Db;
using RestApiServer.Utils;

namespace RestApiServer.Dto.App
{
    public class BoardBasicInfo
    {
        public required BoardEntry Board { get; set; } = null!;
        public required UserBasicInfo CreatedByUser { get; set; } = null!;
    }

    public class BoardFullInfo
    {
        public required BoardEntry Board { get; set; } = null!;
        public required List<TopicBasicInfo> Topics { get; set; } = new();
        public required int TotalTopics { get; set; } = 0;
    }
}